editor/src/components/context-menu-wrapper.tsx

```diff
@@ -18,206 +27,195 @@ import type { WindowPoint } from '../core/shared/math-utils'
-          }
-        })}
-      </Menu>
-    )
-  }
+  return (
+    <Menu key={id} id={id} animation={false} onShown={onShown} onHidden={onHidden}>
+      {splitItems.map((item, index) => {
+        if (item?.type === 'submenu') {
+          return (
+            <SubmenuComponent
+              key={`context-menu-${index}`}
+              label={
+                <span style={{ height: 28, display: 'flex', alignItems: 'center' }}>
+                  {item.label}
+                </span>
+              }
+              arrow={<Icons.ExpansionArrowRightWhite style={{ marginLeft: 8 }} />}
+            >
+              {item.items.map(renderItem)}
+            </SubmenuComponent>
+          )
+        } else {
+          if (item === null) return null
+          return renderItem(item.item, index)
+        }
+      })}
+    </Menu>
+  )
 }
 
-export class ContextMenuWrapper<T> extends ReactComponent<
-  ContextMenuWrapperProps<T> & {
-    dispatch?: EditorDispatch
-    children?: React.ReactNode
-  }
-> {
-  getData = () => this.props.data
-  wrapperStopPropagation = (event: React.MouseEvent<HTMLElement>) => {
-    event.stopPropagation()
-  }
-  render() {
-    const name = `${this.props.id}-context-menu-wrapper`
-    return (
-      <div
-        key={name}
-        className={name + ' ' + (this.props.className ?? '')}
-        style={this.props.style}
-        onMouseDown={this.wrapperStopPropagation}
-        onMouseUp={this.wrapperStopPropagation}
-        onClick={this.wrapperStopPropagation}
-        ref={this.props.forwardRef}
-      >
-        <MenuProvider
-          key={`${this.props.id}-provider`}
-          id={this.props.id}
-          style={this.props.providerStyle}
-          itemsLength={this.props.items.length}
-        >
-          {this.props.children}
-        </MenuProvider>
-        <MomentumContextMenu
-          key={`${this.props.id}`}
-          id={this.props.id}
-          items={this.props.items}
-          dispatch={this.props.dispatch}
-          getData={this.getData}
-        />
-      </div>
-    )
-  }
+export const ContextMenuWrapper = <T,>({
+  children,
+  className = '',
+  data,
+  dispatch,
+  forwardRef,
+  id,
+  items,
+  providerStyle,
+  style,
+}: ContextMenuWrapperProps<T>) => {
+  const name = `${id}-context-menu-wrapper`
+
+  const stopPropagation = useCallback(
+    (event: MouseEvent<HTMLElement>) => event.stopPropagation(),
+    [],
+  )
+
+  const getData = useCallback(() => data, [data])
+
+  return (
+    <div
+      key={name}
+      className={[name, className].join(' ')}
+      style={style}
+      onMouseDown={stopPropagation}
+      onMouseUp={stopPropagation}
+      onClick={stopPropagation}
+      ref={forwardRef}
+    >
+      <MenuProvider id={id} itemsLength={items.length} key={`${id}-provider`} style={providerStyle}>
+        {children}
+      </MenuProvider>
+      <MomentumContextMenu dispatch={dispatch} getData={getData} id={id} items={items} key={id} />
```

Comment:
small thing, but can we rename this to just `ContextMenu` (or `UtopiaContextMenu` as much as that feels redundant)? Momentum was an old product name we used in the mid-1990s

Reply:
Sounds good, I was curious what Momentum meant tbh, didn't really understand!

Reply:
Done in it's own commit